Mathlib/FieldTheory/Adjoin.lean

```diff
@@ -1388,6 +1388,79 @@ end Induction
 
 end IntermediateField
 
+section Minpoly
+
+open AlgEquiv
+
+variable {K L : Type _} [Field K] [Field L] [Algebra K L]
+namespace AdjoinRoot
+
+/-- The canonical algebraic equivalence between `AdjoinRoot p` and `AdjoinRoot q`, where
+  the two polynomial `p q : K[X]` are equal.-/
+noncomputable def id_algEquiv {p q : K[X]} (hp : p ≠ 0) (hq : q ≠ 0) (h_eq : p = q) :
+    AdjoinRoot p ≃ₐ[K] AdjoinRoot q :=
+  ofAlgHom (liftHom p (root q) (by rw [h_eq, aeval_eq, mk_self]))
+    (liftHom q (root p) (by rw [h_eq, aeval_eq, mk_self]))
+    (PowerBasis.algHom_ext (powerBasis hq)
+      (by
+        rw [powerBasis_gen hq, AlgHom.coe_comp, Function.comp_apply, liftHom_root, liftHom_root,
+          AlgHom.coe_id, id.def]))
+    (PowerBasis.algHom_ext (powerBasis hp)
+      (by
+        rw [powerBasis_gen hp, AlgHom.coe_comp, Function.comp_apply, liftHom_root, liftHom_root,
+          AlgHom.coe_id, id.def]))
+
+theorem id_algEquiv_def' {p q : K[X]} (hp : p ≠ 0) (hq : q ≠ 0) (h_eq : p = q) :
+    (id_algEquiv hp hq h_eq).toFun = liftHom p (root q) (by rw [h_eq, aeval_eq, mk_self]) :=
+  rfl
+
+theorem id_algEquiv_def {p q : K[X]} (hp : p ≠ 0) (hq : q ≠ 0) (h_eq : p = q) :
+    (id_algEquiv hp hq h_eq).toAlgHom = liftHom p (root q) (by rw [h_eq, aeval_eq, mk_self]) :=
+  rfl
+
+/-- `id_alg_equiv` sends `adjoin_root.root p` to `adjoin_root.root q`. -/
+theorem id_algEquiv_apply_root {p q : K[X]} (hp : p ≠ 0) (hq : q ≠ 0) (h_eq : p = q) :
+    id_algEquiv hp hq h_eq (root p) = root q := by
+  rw [← coe_algHom, id_algEquiv_def, liftHom_root]
+
+end AdjoinRoot
+
+namespace minpoly
+
+open IntermediateField
+
+/-- If `y : L` is a root of `minpoly K x`, then `minpoly K y = minpoly K x`. -/
+theorem eq_of_root (h_alg : Algebra.IsAlgebraic K L) {x y : L}
+    (h_ev : (Polynomial.aeval y) (minpoly K x) = 0) : minpoly K y = minpoly K x :=
+  Polynomial.eq_of_monic_of_associated (monic (isAlgebraic_iff_isIntegral.mp (h_alg _)))
+    (monic (isAlgebraic_iff_isIntegral.mp (h_alg _)))
+    (Irreducible.associated_of_dvd (irreducible (isAlgebraic_iff_isIntegral.mp (h_alg _)))
+      (irreducible (isAlgebraic_iff_isIntegral.mp (h_alg _))) (dvd K y h_ev))
```

Comment:
Maybe you only need `IsAlgebraic K x` but not the whole `Algebra.IsAlgebraic K L`? Same as the following results.

Also, the `isAlgebraic_iff_isIntegral.mp (h_alg _)` can be rewritten as `(h_alg _).isIntegral` if you like (if you apply my previous suggestion, then it's `h_alg.isIntegral`), it's shorter.

Reply:
Good point, thanks!